internal/worker/worker.go

```diff
@@ -29,13 +29,15 @@ type Worker struct {
 	syncRequested chan bool
 	vmm           *vmmanager.VMManager
 	client        *client.Client
+	pollTicker    *time.Ticker
```

Comment:
Why did you add it?

Reply:
We had no time-based rate limiter in the `Worker.Run()` loop, so I've added it there, and re-used the ticker this way.